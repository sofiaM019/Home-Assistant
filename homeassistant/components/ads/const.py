"""Support for Automation Device Specification (ADS)."""

from __future__ import annotations

from enum import StrEnum
from typing import TYPE_CHECKING

from homeassistant.util.hass_dict import HassKey

if TYPE_CHECKING:
    from .hub import AdsHub

DOMAIN = "ads"

DATA_ADS: HassKey[AdsHub] = HassKey(DOMAIN)
STATE_KEY_STATE = "state"

SERVICE_WRITE_DATA_BY_NAME = "write_data_by_name"

CONF_ADS_VAR = "adsvar"
CONF_ADS_FACTOR = "factor"
CONF_ADS_TYPE = "adstype"
CONF_ADS_VALUE = "value"


class AdsType(StrEnum):
    """Supported Types."""

    BOOL = "bool"
    BYTE = "byte"
    INT = "int"
    UINT = "uint"
    SINT = "sint"
    USINT = "usint"
    DINT = "dint"
    UDINT = "udint"
    WORD = "word"
    DWORD = "dword"
    LREAL = "lreal"
    REAL = "real"
    STRING = "string"
    TIME = "time"
    DATE = "date"
    DATE_AND_TIME = "dt"
    TOD = "tod"
