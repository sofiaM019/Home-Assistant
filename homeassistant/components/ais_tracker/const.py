"""AIS tracker constants."""

DOMAIN = "ais_tracker"

CONF_MMSIS = "mmsis"
DEFAULT_PORT = 12346
