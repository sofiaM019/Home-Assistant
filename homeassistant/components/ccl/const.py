"""Constants for the CCL Electronics integration."""

DOMAIN = "ccl"
