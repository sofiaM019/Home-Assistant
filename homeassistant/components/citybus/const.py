"""CityBus constants"""

DOMAIN = "citybus"

CONF_ROUTE = "route"
CONF_DIRECTION = "direction"
CONF_STOP = "stop"
