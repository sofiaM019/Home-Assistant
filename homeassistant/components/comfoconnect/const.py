"""Constants for the ComfoConnect integration."""

DOMAIN = "comfoconnect"

CONF_USER_AGENT = "user_agent"
DEFAULT_PIN = 0
DEFAULT_TOKEN = "00000000000000000000000000000001"
DEFAULT_USER_AGENT = "Home Assistant"
