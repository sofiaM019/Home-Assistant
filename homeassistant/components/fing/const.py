"""Const for the Fing integration."""

AGENT_IP = "agent_ip"
AGENT_PORT = "agent_port"
AGENT_KEY = "agent_key"
DOMAIN = "fing"
