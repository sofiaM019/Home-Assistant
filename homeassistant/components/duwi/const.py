"""Constants for the DUWI integration."""

from enum import StrEnum
import logging

from homeassistant.const import Platform

_LOGGER = logging.getLogger(__package__)

MANUFACTURER = "duwi"

DOMAIN = "duwi"

APP_VERSION = "0.1.1"

CLIENT_VERSION = "0.1.1"

CLIENT_MODEL = "homeassistant"

DUWI_DISCOVERY_NEW = "duwi_discovery_new"
DUWI_SCENE_UPDATE = "duwi_scene_update"
DUWI_HA_SIGNAL_UPDATE_ENTITY = "duwi_entry_update"
DUWI_HA_ACCESS_TOKEN = "duwi_access_token"

# API keys
CLIENT = "client"
ADDRESS = "address"
WS_ADDRESS = "ws_address"
APP_KEY = "app_key"
APP_SECRET = "app_secret"
ACCESS_TOKEN = "access_token"
REFRESH_TOKEN = "refresh_token"
PHONE = "phone"
PASSWORD = "password"
HOUSE_NO = "house_no"
HOUSE_NAME = "house_name"
HOUSE_KEY = "house_key"
CONF_TOKEN_INFO = "token_info"

# List of platforms that support config entry
SUPPORTED_PLATFORMS = [
    Platform.SWITCH,
]


class DPCode(StrEnum):
    """Data Point Codes used by Duwi."""

    # button
    BUTTON = "button"

    # switch
    SWITCH = "switch"

    # light
    ON_OFF = "on_off"
    LIGHT = "light"
    COLOR_TEMP = "color_temp"
    COLOR = "color"
    RGB = "rgb"
    RGBW = "rgbw"
    RGBCW = "rgbcw"

    # cover
    ROLLER = "roller"
    SHUTTER = "shutter"
    SHUTTER_2 = "shutter-2"

    # music
    HUA_ERSI_MUSIC = "hua_ersi_music"
    XIANG_WANG_MUSIC_S7_MINI_3S = "xiangwang_music_s7_mini_3s"
    XIANG_WANG_MUSIC_S8 = "xiangwang_music_s8"
    SHENG_BI_KE_MUSIC = "sheng_bike_music"
    BO_SHENG_MUSIC = "bo_sheng_music"
    YOU_DA_MUSIC = "youda_music"

    # sensor
    TEMP = "temp"
    HUMIDITY = "humidity"
    HCHO = "hcho"
    PM25 = "pm25"
    CO2 = "co2"
    BRIGHT = "bright"
    IQA = "iaq"
    HUMAN = "human"
    TRIGGER = "trigger"
    MOISTURE = "moisture"
    SMOKE = "smoke"
    MOVING = "moving"
    GAS = "gas"
    DOOR = "door"
    OPENING = "opening"
    CO = "co"
    TVOC = "tvoc"
    PM10 = "pm10"

    # climate
    AC = "ac"
    FA = "fa"
    FH = "fh"
    HP = "hp"
    TC = "tc"
