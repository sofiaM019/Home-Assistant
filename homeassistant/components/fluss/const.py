"""Constants for the Fluss+ integration."""

from homeassistant.const import CONF_API_KEY  # noqa: F401

DOMAIN = "fluss"
BASE_URL = "https://zgekzokxrl.execute-api.eu-west-1.amazonaws.com/v1/api/"

CONF_BASE_URL = BASE_URL  # You can refer to BASE_URL this way
