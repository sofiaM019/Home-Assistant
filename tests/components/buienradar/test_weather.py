"""The tests for the buienradar weather component."""

from http import HTTPStatus

from syrupy.assertion import SnapshotAssertion

from homeassistant.components.buienradar.const import DOMAIN
from homeassistant.components.weather import (
    DOMAIN as WEATHER_DOMAIN,
    SERVICE_GET_FORECASTS,
)
from homeassistant.const import ATTR_ENTITY_ID, CONF_LATITUDE, CONF_LONGITUDE, CONF_TYPE
from homeassistant.core import HomeAssistant
from homeassistant.setup import async_setup_component

from tests.common import MockConfigEntry, load_fixture, load_json_object_fixture
from tests.test_util.aiohttp import AiohttpClientMocker

TEST_CFG_DATA = {CONF_LATITUDE: 51.5288504, CONF_LONGITUDE: 5.4002156}
TEST_CFG_INVALID_COORDINATES = {CONF_LATITUDE: None, CONF_LONGITUDE: None}
TEST_CFG_INVALID_DATA = {CONF_LATITUDE: 53.45, CONF_LONGITUDE: 6.42}
ENTITY_ID = "weather.buienradar"


async def test_smoke_test_setup_component(
    aioclient_mock: AiohttpClientMocker, hass: HomeAssistant
) -> None:
    """Smoke test for successfully set-up with default config."""
    aioclient_mock.get(
        "https://data.buienradar.nl/2.0/feed/json", status=HTTPStatus.NOT_FOUND
    )

    mock_entry = MockConfigEntry(domain=DOMAIN, unique_id="TEST_ID", data=TEST_CFG_DATA)

    mock_entry.add_to_hass(hass)

    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    state = hass.states.get("weather.buienradar")
    assert state.state == "unknown"


async def test_smoke_test_setup_component_no_coordinates(
    aioclient_mock: AiohttpClientMocker, hass: HomeAssistant
) -> None:
    """Smoke test for set-up without valid coordinates."""
    aioclient_mock.get(
        "https://data.buienradar.nl/2.0/feed/json", status=HTTPStatus.NOT_FOUND
    )

    mock_entry = MockConfigEntry(
        domain=DOMAIN, unique_id="TEST_ID", data=TEST_CFG_INVALID_COORDINATES
    )

    mock_entry.add_to_hass(hass)

    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    state = hass.states.get("weather.buienradar")
    assert state is None


async def test_attribute_values(
    aioclient_mock: AiohttpClientMocker,
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
) -> None:
    """Test for fetching (mocked) weather data and checking attribute values."""
    brJson = load_json_object_fixture("buienradar.json", DOMAIN)
    raindata = load_fixture("raindata.txt", DOMAIN)
    aioclient_mock.get(
        "https://data.buienradar.nl/2.0/feed/json", status=HTTPStatus.OK, json=brJson
    )
    aioclient_mock.get(
        f"https://gpsgadget.buienradar.nl/data/raintext?lat={TEST_CFG_DATA[CONF_LATITUDE]:.2f}&lon={TEST_CFG_DATA[CONF_LONGITUDE]:.1f}",
        status=HTTPStatus.OK,
        text=raindata,
    )

    mock_entry = MockConfigEntry(domain=DOMAIN, unique_id="TEST_ID", data=TEST_CFG_DATA)

    mock_entry.add_to_hass(hass)

    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    state = hass.states.get("weather.buienradar")
    assert state == snapshot


async def test_get_forecast(
    aioclient_mock: AiohttpClientMocker,
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
) -> None:
    """Test for fetching (mocked) weather data and checking forecast values."""
    brJson = load_json_object_fixture("buienradar.json", DOMAIN)
    raindata = load_fixture("raindata.txt", DOMAIN)
    aioclient_mock.get(
        "https://data.buienradar.nl/2.0/feed/json", status=HTTPStatus.OK, json=brJson
    )
    aioclient_mock.get(
        f"https://gpsgadget.buienradar.nl/data/raintext?lat={TEST_CFG_DATA[CONF_LATITUDE]:.2f}&lon={TEST_CFG_DATA[CONF_LONGITUDE]:.1f}",
        status=HTTPStatus.OK,
        text=raindata,
    )

    mock_entry = MockConfigEntry(domain=DOMAIN, unique_id="TEST_ID", data=TEST_CFG_DATA)
    mock_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_entry.entry_id)

    assert await async_setup_component(hass, DOMAIN, {})
    await hass.async_block_till_done()

    response = await hass.services.async_call(
        WEATHER_DOMAIN,
        SERVICE_GET_FORECASTS,
        {ATTR_ENTITY_ID: ENTITY_ID, CONF_TYPE: "daily"},
        blocking=True,
        return_response=True,
    )
    await hass.async_block_till_done()

    assert response == snapshot


async def test_attribute_values_invalid_data(
    aioclient_mock: AiohttpClientMocker,
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
) -> None:
    """Test for fetching (mocked) weather data (with some values not present/invalid) to test attribute values."""
    brJson = load_json_object_fixture("buienradar.json", DOMAIN)
    raindata = load_fixture("raindata.txt", DOMAIN)
    aioclient_mock.get(
        "https://data.buienradar.nl/2.0/feed/json", status=HTTPStatus.OK, json=brJson
    )
    aioclient_mock.get(
        f"https://gpsgadget.buienradar.nl/data/raintext?lat={TEST_CFG_INVALID_DATA[CONF_LATITUDE]:.2f}&lon={TEST_CFG_INVALID_DATA[CONF_LONGITUDE]:.2f}",
        status=HTTPStatus.OK,
        text=raindata,
    )

    mock_entry = MockConfigEntry(
        domain=DOMAIN, unique_id="TEST_ID", data=TEST_CFG_INVALID_DATA
    )

    mock_entry.add_to_hass(hass)

    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    state = hass.states.get("weather.buienradar")
    assert state == snapshot


async def test_smoke_test_no_forecasts(
    aioclient_mock: AiohttpClientMocker,
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
) -> None:
    """Test for fetching (mocked) weather data (with forecast data missing) to test attribute values."""
    brJson = load_json_object_fixture("buienradarNoForecast.json", DOMAIN)
    raindata = load_fixture("raindata.txt", DOMAIN)
    aioclient_mock.get(
        "https://data.buienradar.nl/2.0/feed/json", status=HTTPStatus.OK, json=brJson
    )
    aioclient_mock.get(
        f"https://gpsgadget.buienradar.nl/data/raintext?lat={TEST_CFG_INVALID_DATA[CONF_LATITUDE]:.2f}&lon={TEST_CFG_INVALID_DATA[CONF_LONGITUDE]:.2f}",
        status=HTTPStatus.OK,
        text=raindata,
    )

    mock_entry = MockConfigEntry(
        domain=DOMAIN, unique_id="TEST_ID", data=TEST_CFG_INVALID_DATA
    )

    mock_entry.add_to_hass(hass)

    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    state = hass.states.get("weather.buienradar")
    assert state == snapshot
