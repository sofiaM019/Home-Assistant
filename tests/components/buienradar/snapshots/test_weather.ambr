# serializer version: 1
# name: test_attribute_values
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'apparent_temperature': 20.5,
      'attribution': 'Data provided by buienradar.nl',
      'friendly_name': 'Buienradar',
      'humidity': 76,
      'precipitation_unit': <UnitOfPrecipitationDepth.MILLIMETERS: 'mm'>,
      'pressure': 1020.0,
      'pressure_unit': <UnitOfPressure.HPA: 'hPa'>,
      'supported_features': <WeatherEntityFeature: 1>,
      'temperature': 20.4,
      'temperature_unit': <UnitOfTemperature.CELSIUS: '°C'>,
      'visibility': 32.6,
      'visibility_unit': <UnitOfLength.KILOMETERS: 'km'>,
      'wind_bearing': 38,
      'wind_gust_speed': 23.4,
      'wind_speed': 14.4,
      'wind_speed_unit': <UnitOfSpeed.KILOMETERS_PER_HOUR: 'km/h'>,
    }),
    'context': <ANY>,
    'entity_id': 'weather.buienradar',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'cloudy',
  })
# ---
# name: test_attribute_values_invalid_data
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by buienradar.nl',
      'friendly_name': 'Buienradar',
      'precipitation_unit': <UnitOfPrecipitationDepth.MILLIMETERS: 'mm'>,
      'pressure_unit': <UnitOfPressure.HPA: 'hPa'>,
      'supported_features': <WeatherEntityFeature: 1>,
      'temperature_unit': <UnitOfTemperature.CELSIUS: '°C'>,
      'visibility_unit': <UnitOfLength.KILOMETERS: 'km'>,
      'wind_speed_unit': <UnitOfSpeed.KILOMETERS_PER_HOUR: 'km/h'>,
    }),
    'context': <ANY>,
    'entity_id': 'weather.buienradar',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'cloudy',
  })
# ---
# name: test_get_forecast
  dict({
    'weather.buienradar': dict({
      'forecast': list([
        dict({
          'condition': 'partlycloudy',
          'datetime': '2024-08-31T00:00:00+02:00',
          'precipitation': 4.0,
          'precipitation_probability': 30,
          'temperature': 25.0,
          'templow': 15.0,
          'wind_bearing': 45,
          'wind_speed': 20.38,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2024-09-01T00:00:00+02:00',
          'precipitation': 3.0,
          'precipitation_probability': 30,
          'temperature': 29.0,
          'templow': 17.0,
          'wind_bearing': 90,
          'wind_speed': 12.96,
        }),
        dict({
          'condition': 'pouring',
          'datetime': '2024-09-02T00:00:00+02:00',
          'precipitation': 3.5,
          'precipitation_probability': 60,
          'temperature': 27.0,
          'templow': 18.0,
          'wind_bearing': 225,
          'wind_speed': 7.42,
        }),
        dict({
          'condition': 'rainy',
          'datetime': '2024-09-03T00:00:00+02:00',
          'precipitation': 1.5,
          'precipitation_probability': 40,
          'temperature': 26.0,
          'templow': 17.0,
          'wind_bearing': 270,
          'wind_speed': 12.96,
        }),
        dict({
          'condition': 'rainy',
          'datetime': '2024-09-04T00:00:00+02:00',
          'precipitation': 1.0,
          'precipitation_probability': 40,
          'temperature': 25.0,
          'templow': 14.0,
          'wind_bearing': 180,
          'wind_speed': 7.42,
        }),
      ]),
    }),
  })
# ---
# name: test_smoke_test_no_forecasts
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by buienradar.nl',
      'friendly_name': 'Buienradar',
      'precipitation_unit': <UnitOfPrecipitationDepth.MILLIMETERS: 'mm'>,
      'pressure_unit': <UnitOfPressure.HPA: 'hPa'>,
      'supported_features': <WeatherEntityFeature: 1>,
      'temperature_unit': <UnitOfTemperature.CELSIUS: '°C'>,
      'visibility_unit': <UnitOfLength.KILOMETERS: 'km'>,
      'wind_speed_unit': <UnitOfSpeed.KILOMETERS_PER_HOUR: 'km/h'>,
    }),
    'context': <ANY>,
    'entity_id': 'weather.buienradar',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'cloudy',
  })
# ---
